sdk/identity/identity/samples/AzureCacheForRedis/ioredis.md

```diff
@@ -0,0 +1,273 @@
+
+```ts
+import Redis from "ioredis";
+import { DefaultAzureCredential } from "@azure/identity";
+import * as dotenv from "dotenv";
+dotenv.config();
+
+async function main() {
+  // Construct a Token Credential from Identity library, e.g. ClientSecretCredential / ClientCertificateCredential / ManagedIdentityCredential, etc.
+  const credential = new DefaultAzureCredential();
+
+  // The scope will be changed for Azure AD Public Preview
+  const redisScope = "https://*.cacheinfra.windows.net:10225/appid/.default"
+  
+  // Fetch an Azure AD token to be used for authentication. This token will be used as the password.
+  let accessTokenObject = await credential.getToken(
+    redisScope
+  );
+
+  // Create ioredis client and connect to the Azure Cache for Redis over the TLS port using the access token as password.
+  const redis = new Redis({
+    username: process.env.REDIS_SERVICE_PRINCIPAL_NAME,
+    password: accessTokenObject.token,
+    tls: {
+      host: process.env.REDIS_HOSTNAME,
+      port: 6380,
+    },
+    keepAlive: 0
+  });
+
+  // Set a value against your key in the Azure Redis Cache.
+  await redis.set("Az:mykey", "value123"); // Returns a promise which resolves to "OK" when the command succeeds.
+  // Fetch value of your key in the Azure Redis Cache.
+  console.log("redis key:", await redis.get("Az:mykey"));
+}
+main().catch((err) => {
+  console.log("error code: ", err.code);
+  console.log("error message: ", err.message);
+  console.log("error stack: ", err.stack);
+});
+```
+
+##### Supported Token Credentials for Azure AD Authentication
+
+**Note:** The samples in this doc use the Azure Identity library's `DefaultAzureCredential` to fetch an Azure AD access token. The samples also use the service principal name as the username. The other supported `TokenCredential` implementations that can be used from the [Azure Identity for JavaScript](https://docs.microsoft.com/javascript/api/overview/azure/identity-readme?view=azure-node-latest) library are as follows:
+
+* [Client Certificate Credential](https://docs.microsoft.com/javascript/api/@azure/identity/clientcertificatecredential?view=azure-node-latest)
+* [Client Secret Credential](https://docs.microsoft.com/javascript/api/@azure/identity/clientsecretcredential?view=azure-node-latest)
+* [Managed Identity Credential](https://docs.microsoft.com/javascript/api/@azure/identity/managedidentitycredential?view=azure-node-latest)
+* [Username Password Credential](https://docs.microsoft.com/javascript/api/@azure/identity/usernamepasswordcredential?view=azure-node-latest)
+* [Azure CLI Credential](https://docs.microsoft.com/javascript/api/@azure/identity/azureclicredential?view=azure-node-latest)
+* [Interactive Browser Credential](https://docs.microsoft.com/javascript/api/@azure/identity/interactivebrowsercredential?view=azure-node-latest)
+* [Device Code Credential](https://docs.microsoft.com/javascript/api/@azure/identity/devicecodecredential?view=azure-node-latest)
```

Comment:
Isn't this pretty much all of them? Why would a particular token credential _not_ be supported as long as the access token identifies a principal with the correct role binding?

Reply:
Authorization Code is not included in the list

Reply:
I guess my question is really why those don't work. Auth Code I kind of understand, and VS Code credential is broken, but why not powershell? Could we just say "these three credentials can't be used" and point people to AzureApplicationCredential as a good default?

Reply:
For private preview, these 3 credentials are not supported. For public preview, service may add the support